apt-hook/json-hook.cc

```diff
@@ -275,11 +361,15 @@ int run()
         }
     } else if (hook_req.method == "org.debian.apt.hooks.install.pre-prompt") {
         // ESM stats
-        process_all_templates();
-        std::ifstream esm_message_file(APT_PRE_INVOKE_MESSAGE_STATIC_PATH);
-        if (esm_message_file.is_open()) {
-            std::cout << esm_message_file.rdbuf();
-            esm_message_file.close();
+        ESMUpdates esm_updates;
+        bool success = get_potential_esm_updates(esm_updates);
+        if (success) {
+            ESMContext esm_context = get_esm_context();
+            if (!esm_updates.infra_packages.empty()) {
+                print_esm_packages("esm-infra", esm_updates.infra_packages, esm_context);
+            } else if (!esm_updates.apps_packages.empty()) {
+                print_esm_packages("esm-apps", esm_updates.infra_packages, esm_context);
+            }
```

Comment:
we might need a smarter "are we EOL" check here instead
we can parse distro-info to get that I think

Reply:
from out of band conversation: a smarter eol check isn't needed here because the infra source is only added to the separate apt cache on eol machines